extension-requests/script.js

```diff
@@ -45,24 +50,99 @@ const getUser = async (username) => {
     return user;
   }
 };
+
 const initializeUserMap = (userList) => {
   userList.forEach((user) => {
     userMap.set(user?.username?.toLowerCase(), {
       first_name: user.first_name,
       picture: { url: user.picture?.url },
       id: user.id,
+      username: user.username,
     });
   });
 };
+
+const processUsernames = async (usernames) => {
+  const validUsernameList = [];
+  const userIdList = [];
+  if (!usernames)
+    return {
+      validUsernameList,
+      userIdList,
+    };
+  const usernameList = Array.isArray(usernames)
+    ? usernames
+    : usernames.split(',');
+  const userPromise = [];
+  for (const username of usernameList) {
+    userPromise.push(getUser(username));
+  }
+  const userList = await Promise.all(userPromise);
+  for (const user of userList) {
+    if (user) {
+      validUsernameList.push(user.username);
+      userIdList.push(user.id);
+    }
+  }
+  return { validUsernameList, userIdList };
+};
+
 const initializeUserStatusMap = (userStatusList) => {
   userStatusList.forEach((status) => {
     userStatusMap.set(status.userId, status);
   });
 };
+
+const updateUIBasedOnFilterStates = () => {
+  const states = { ...filterStates };
+  states.assignee = assigneeUsernamesList;
+  if (states.assignee && states.assignee.length > 0) {
+    searchElement.value = states.assignee.join(',');
+  } else {
+    searchElement.value = '';
+  }
+
+  if (states.order === 'asc') {
+    ascIcon.style.display = 'block';
+    descIcon.style.display = 'none';
+  } else if (states.order === 'desc') {
+    ascIcon.style.display = 'none';
+    descIcon.style.display = 'block';
+  }
+
+  if (
+    states.status &&
+    Array.isArray(states.status) &&
+    states.status.length > 0
+  ) {
+    states.status.forEach((state) => {
+      toggleStatusCheckbox(state);
+    });
+  } else {
+    toggleStatusCheckbox(states.status);
+  }
+};
+
 const render = async () => {
   addTooltipToSortButton();
-  toggleStatusCheckbox(Status.PENDING);
+  if (window.location.search) {
+    parseExtensionRequestParams(window.location.search, filterStates);
+    const usersList = await processUsernames(filterStates.assignee);
+    const userIdList = usersList.userIdList;
+    assigneeUsernamesList = usersList.validUsernameList;
+    const assigneeFilterState = userIdList.length ? userIdList : '';
+    filterStates.assignee = assigneeFilterState;
+    if (!filterStates.order) {
+      filterStates.order = Order.ASCENDING;
+    }
+  } else {
+    filterStates.status = Status.PENDING;
```

Comment:
why status is pending if there is no order selected

Reply:
Requirements